### PR TITLE
Add commands to set color profile and move between profiles

### PR DESCRIPTION
**Pull request by tech2077**
The main motivation for these commands is for persistent color profiles and led state or for keyboard layouts with separate on/off and change profile keys.

This firmware does change the behavior of the CMD_LED_ON and CMD_LED_OFF command to prevent cycling the profile when loading the led state from persistent memory.

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -169,7 +207,7 @@ void executeProfile(){
  * Turn off all leds
  */
 void disableLeds(){
-  currentProfile = (currentProfile+amountOfProfiles-1)%amountOfProfiles;
+  // currentProfile = (currentProfile+amountOfProfiles-1)%amountOfProfiles;
```
Comment (Codetector1374) [inline main.c:210]: please just remove this line altogether. Tying to not have a lot of "commented out" code.

File: source/ap2_qmk_led.h [hunk 2]
```diff
@@ -3,4 +3,9 @@
...
 #define CMD_LED_OFF                         0x2
 #define CMD_LED_SET                         0x3
-#define CMD_LED_SET_ROW                     0x4
\ No newline at end of file
+#define CMD_LED_SET_ROW                     0x4
+#define CMD_LED_SET_PROFILE                 0x5
+#define CMD_LED_NEXT_PROFILE                0x6
+#define CMD_LED_PREV_PROFILE                0x7
+#define CMD_LED_GET_PROFILE                 0x8
+#define CMD_LED_GET_NUM_PROFILES            0x9
```
Comment (Codetector1374) [inline source/ap2_qmk_led.h:11]: New line at EOF

File: main.c [hunk 3]
```diff
@@ -113,9 +114,9 @@ static uint8_t commandBuffer[64];
  * Thread 1.
  */
 THD_WORKING_AREA(waThread1, 128);
-THD_FUNCTION(Thread1, arg) {
+_Noreturn THD_FUNCTION(Thread1, arg) {
```
Comment (Codetector1374) [inline main.c:117]: consider using ```__attribute__((noreturn))```

File: main.c [hunk 4]
```diff
@@ -277,8 +315,9 @@ int main(void) {
   chSysInit();
 
   // Setup UART1
+  profiles[currentProfile](ledColors);
```
Comment (Codetector1374) [inline main.c:318]: keep the "Setup UART1" comment next to the UART1 setup section (line 320) (Or remove it it seems kind of redundant anyway)

**Comment by tech2077** [comment]
Added in requested changes. I'll get the related qmk changes cherry-picked into a PR so they can be merged along side these changes.

**Comment by Codetector1374** [comment]
lgtm

**Pull request merged**